src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -857,11 +1104,83 @@ void ProtocolHandlerImpl::OnConnectionClosed(
   multiframe_builder_.RemoveConnection(connection_id);
 }
 
+void ProtocolHandlerImpl::OnUnexpectedDisconnect(
+    const transport_manager::ConnectionUID connection_id,
+    const transport_manager::CommunicationError& error) {
+  OnConnectionClosed(connection_id);
+}
+
 void ProtocolHandlerImpl::NotifyOnFailedHandshake() {
   LOG4CXX_AUTO_TRACE(logger_);
   security_manager_->NotifyListenersOnHandshakeFailed();
 }
 
+void ProtocolHandlerImpl::OnTransportConfigUpdated(
+    const transport_manager::transport_adapter::TransportConfig& configs) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  transport_manager::transport_adapter::TransportConfig::const_iterator it =
+      configs.find("enabled");
+  if (configs.end() == it) {
+    LOG4CXX_WARN(logger_, "No enabled field in OnTransportConfigUpdated");
+    return;
+  }
+
+  bool tcp_enabled = (0 == strcmp("true", it->second.c_str()));
+  std::string tcp_port;
+
+  if (tcp_enabled) {
+    it = configs.find("tcp_port");
+    if (configs.end() == it) {
+      LOG4CXX_WARN(logger_, "No port field in OnTransportConfigUpdated");
+      return;
+    }
+    tcp_port = it->second;
+
+    it = configs.find("tcp_ip_address");
+    if (configs.end() == it) {
+      LOG4CXX_WARN(logger_, "No IP address field in OnTransportConfigUpdated");
+      return;
+    }
+    tcp_enabled_ = true;
+    tcp_port_ = tcp_port;
+    tcp_ip_address_ = it->second;
+  } else {
+    tcp_enabled_ = false;
+    tcp_port_.clear();
+    tcp_ip_address_.clear();
+  }
+
+  LOG4CXX_INFO(logger_,
+               "OnTransportConfigUpdated: new config enabled is "
+                   << tcp_enabled_ << ". Port is " << tcp_port_
+                   << ". IP Address is " << tcp_ip_address_);
+
+  // Walk the SessionConnection map and find all sessions that need a
+  // TransportUpdate Event. Sessions flagged with 0xFFFFFFFF in their
+  // secondary transport are ineligible for secondary transport, and
+  // therefore don't get this event.
+  NonConstDataAccessor<connection_handler::SessionConnectionMap>
```

Comment:
@jacobkeeler We probably don't need the NonConst version here, but we do need it in ConnectionHandlerImpl::SetSecondaryTransportID, where we set the secondary connection ID for an existing connection in the map.

As for this NonConstDataAccessor, I wrote this for another PR way back in December that hasn't been approved yet (#1958)

Reply:
You don't need that in `ConnectionHandlerImpl::SetSecondaryTransportID`, because you have access to the underlying lock there, and it is used manually in several other parts of the class as well.

Reply:
I don't believe it is necessary for this PR, so I would like to request that it be removed for now.

Reply:
We removed the usage of `NonConstDataAccessor` with commit https://github.com/smartdevicelink/sdl_core/pull/2164/commits/08acb77b6b5a43116a0008905aeaf960fd2e09f5 and https://github.com/smartdevicelink/sdl_core/pull/2164/commits/cf65cd85f837ad43db7609afd3e089045c22f980.